### ISSUE TITLE
Add DeviceType as Info / Feature

### ISSUE THREAD

**Issue by dfarin** [blakadder/zigbee#1051]
Hi, sorry for my terrible noob posting. :)
It would be awesome to have the Device Type as an info.
As an Zigbee newbie i bought 6 switches now with the knowledge, almost every wired device (switch bulb etc) is an router, but my expirience is , its not :( 
All switches i bought, aqara, moes, sonoff are just EndDevice, so it would be great to have such an info in your really wonderful database.

What do you think ?

:)

**Comment by blakadder** [comment]
how do you propose to acquire that information?

**Comment by dfarin** [comment]
> how do you propose to acquire that information?
With people who already have those stuff ? As additional user info or similar ?
I think when there is a place for, people share this with you.

Like me, i got several things now and i was, not angry, but disapointed, because i have 4 switches i "cant use" for what i need them .

So right now it would be very easy for me to share this information. But before i just edit/add such infos i think asking is the better way.
i mean some very simple thing like:
![image](https://github.com/blakadder/zigbee/assets/9096265/85011ef9-05b5-4504-ba9d-ed77756806f4)

**Comment by blakadder** [comment]
Yes cool. How do you propose to gather and verify that information for 2500+ devices?

**Comment by dfarin** [comment]
Do you realy need every information instant for every device ?
What about just let it grow for later users who need this ?
Verify is a problem of course, but thats why i call it "user content" so everybody can decide for themself if he trust this or not.

**Comment by blakadder** [comment]
Any mains powered device serves as a Zigbee router unless the manufacturer decided against that, some devices in the database already include a note that they don't work as a router so the user content is already there

**Comment by dfarin** [comment]
Yes, i found some with that info but not as a seperate point, so most times i cant see if its just a missing info or if its your logic "Any mains powered device serves as a Zigbee router " and thats very confusing.
I bought 5 different devices now were there was no info and then the logic "wired is router" was clear for me, but as i got them and try to test, i saw, oh, wired is enddevice.
Of course its a manufacturer problem that they dont share such infos but i thought that espacily for those cases an independent database is the best use case and solution. And for a solution it is necessary to see "missing info" or similar.

And yes, its a lot, but for sure not for +2500 because not every device is wired

And yes, forgive me terrible english, i try my best :)

**Comment by gkawanami** [comment]
@dfarin 
You may want to avoid "no neutral" devices, because most of them are end- devices. ('Cause of the power consumption thing)
As for this kind of info being available, we have to understand that @blakadder has a history  of gathering that kind of info for a long time, the guy owns  one of the biggest (if not the biggest) resource pages for that kind of data.  If someone knows the complexity of gathering and verifying manufacturer's informations and compliance of these manufacturers to the protocol, its him.

**Comment by blakadder** [comment]
> Yes, i found some with that info but not as a seperate point, so most times i cant see if its just a missing info or if its your logic "Any mains powered device serves as a Zigbee router " and thats very confusing. 

That is not my logic, that is how things are.

>I bought 5 different devices now were there was no info and then the logic "wired is router" was clear for me, but as i got them and try to test, i saw, oh, wired is enddevice. Of course its a manufacturer problem that they dont share such infos but i thought that espacily for those cases an independent database is the best use case and solution. And for a solution it is necessary to see "missing info" or similar.

I didn't say wired, I said "mains powered"
> 
> And yes, its a lot, but for sure not for +2500 because not every device is wired

Ok, but even if its 1000 doesn't make it a quick task, does it?
> 
> And yes, forgive me terrible english, i try my best :)

Anyway, I don't see why the drama. If you have information submit it to the device page as you've seen done with other devices.

**Comment by aleksei-oecd** [comment]
Hi, just discovered this topic looking for the same information as a topic starter. I was looking for a wall switch that can act as a router to make a mesh more robust as it is the most common component. And yes it's a pity that this information is not provided either by the manufacturer or in independent databases. From hard experience (buying + waiting for a delivery + installing) some like MOES, BSEED, Aqara with neutral lines discovered that they all end-devices.  So an extra field in DB for a device type [router/end-device/unknown] would be a very useful thing.

**Comment by dfarin** [comment]
> Hi, just discovered this topic looking for the same information as a topic starter. I was looking for a wall switch that can act as a router to make a mesh more robust as it is the most common component. And yes it's a pity that this information is not provided either by the manufacturer or in independent databases. From hard experience (buying + waiting for a delivery + installing) some like MOES, BSEED, Aqara with neutral lines discovered that they all end-devices. So an extra field in DB for a device type [router/end-device/unknown] would be a very useful thing.

My experience is, i use several plugs (around 10 Euro), some USB Repeater (also around 10 Euro) and right now im testing wall outlets. Best way is using "bulbs" and dont use the classic switch anymore. I put some push buttons nearby the classic switches so my wife can use it almost classic.

**Comment by pipiche38** [comment]
Main powered devices without neutral are usually not router

**Comment by stretch4x4** [comment]
I would certainly appreciate this feature.
I need to get a device that can improve my network as several devices can't connect reliably.
Being able to search by router based devices would help me to find a device that can improve my connectivity and do something else useful for me.
Another useful search parameter for me, would be devices that have an AU power plug.
